[Ren Weilin] iP

wlren opened the pull request:
### DukeNoob
> “Sometimes when I close my eyes, I can't see.” – ??? ([source](https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.coolfunnyquotes.com%2Fcategory%2Fstupid-funny-quotes%2F&psig=AOvVaw2bMgXjB6xRTXYUR90SRsQD&ust=1631272658388000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCNjL06zi8fICFQAAAAAdAAAAABAD))

DukeNoob helps your mind with forgetting things you need to do. It's,

- meme-based
- hard to learn
- ~~FAST~~ _SUPER SLOW_ to use

All you need to do is,

1. download it from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ).
2. double-click it.
3. add your tasks.
4. let it delete your tasks for you 😉

And it is only **99.99 SGD**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

Here's a snippet of my awesome `code`:

```java
    /**
     * Deletes task at index from the list.
     * 
     * @param index
     */
    public void deleteTask(int index) {
        tasks.remove(index);
    }
```

Commit:
Add Gradle support

wlren committed:
Level-1 Logic (Greet, Echo, Exit)

wlren committed:
Level-2 (Add, List)

wlren committed:
Level-3 (Mark as Done)

wlren committed:
Level-4 (Todos, Events, Deadlines)

wlren committed:
Level-5 Basic Error handling

wlren committed:
First iteration of tests

wlren committed:
Delete functionality

wlren committed:
Save functionality

wlren committed:
Update .gitignore

wlren committed:
Date support

wlren committed:
Merge branch 'branch-Level-7'

wlren committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

wlren committed:
Enum for Command

wlren committed:
Restructure project

wlren committed:
Duke Package

wlren committed:
First test cases

wlren committed:
JAR Build

wlren committed:
Find functionality

wlren committed:
Add Java Doc for most methods

wlren committed:
Minor changes for code standard

wlren committed:
Merge branch 'branch-Level-9'

wlren committed:
Merge branch 'branch-A-CodingStandard'

wlren committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

kheekheekhee commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,31 @@
+ * @author: Ren Weilin A0216723J
+ */
+public class Duke {
+    private final UI userInterface;
```
Should `UI` be spelt `Ui` instead?

kheekheekhee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,97 @@
+     *
+     * @return CommandType The command type of the next line of input
+     */
+    public CommandType nextCommand() {
```
Should `nextCommand` be a verb? Perhaps you could use `getNextCommand` instead?

kheekheekhee commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,72 @@
+        tasks.remove(index);
+    }
+
+    /**
+     * Marks the task at index as done.
+     *
+     * @param index
+     */
+    public void markAsDone(int index) {
+        tasks.get(index).markAsDone();
+    }
+
+    /**
+     * Retrieves the size of task list.
+     *
+     * @return int Size of task list.
+     */
+    public int getListSize() {
+        return tasks.size();
+    }
+
+    /**
+     * Gets the string representation of task at index.
+     *
+     * @param index
+     * @return String toString() of task at index.
+     */
+    public String getTaskString(int index) {
+        return tasks.get(index).toString();
+    }
+
+    /**
+     * Exposes the task list for operations.
+     *
+     * @return ArrayList<Task> Duke's task list.
+     */
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+}
```
Very clean code here. Well done! 👍

kheekheekhee commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,158 @@
+
+    private void addTaskMsg(Task newTask) {
+        printBreakLine();
+        System.out.println("  Got it. I've added this task: ");
```
Any reason you added spaces instead of using \t for a tab indentation?

kheekheekhee commented on src/main/java/duke/CommandType.java:
```diff
@@ -0,0 +1,16 @@
+public enum CommandType {
+    Exit,
+    List,
+    MarkAsDone,
+    DeleteTask,
+    Find,
+    Error,
+    AddToDo,
+    AddDeadline,
+    AddEvent
```
Should enum names be nouns instead of verbs? Perhaps `Event` instead of `AddEvent`.

kheekheekhee reviewed (commented):
LGTM! Couldn't find any major errors aside from some minor nitpicks and small violations of naming conventions. Keep up the good work!

bernarduskrishna commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    /**
+     * String representation of deadline task.
```
You can look at the Javadoc convention. "In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returnning etc.)"

bernarduskrishna commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,31 @@
+    /**
+     * Start of Duke chatbot.
+     *
+     * @param args
```
Should you give description for args?

bernarduskrishna commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,97 @@
+        case "find":
+            return CommandType.Find;
+        default:
+            return CommandType.Error;
```
All these commands are constants so possibly you can make an enum?

bernarduskrishna replied:
Possibly issue for some other switch case uses also

bernarduskrishna commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,89 @@
+     */
+    public Storage(TaskList taskList) {
+        this.taskList = taskList;
+        String DIR_NAME = "data";
+        String FILE_NAME = "duke.txt";
```
Possibly declare final?

bernarduskrishna commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,89 @@
+                description = taskString.substring(7) + " ";
+                taskList.addTask(new ToDo(description, isDone));
+                break;
+            case 'E':
+                timeStart = taskString.indexOf("at: ");
+                description = taskString.substring(7, timeStart - 2) + " ";
+                dateTime = taskString.substring(timeStart + 4, taskString.length() - 1);
+                date = LocalDate.parse(dateTime, DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM));
+                taskList.addTask(new Event(description, date, isDone));
+                break;
+            case 'D':
+                timeStart = taskString.indexOf("by: ");
+                description = taskString.substring(7, timeStart - 2) + " ";
+                dateTime = taskString.substring(timeStart + 4, taskString.length() - 1);
+                date = LocalDate.parse(dateTime, DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM));
+                taskList.addTask(new Deadline(description, date, isDone));
```
Possibly can abstract these chunks out as they seem to be quite similar. Some other parts also can be abstracted better I think.

bernarduskrishna reviewed (commented):
In general, good job! I think a few things that the code can be improved on are the javadocs and the abstraction. As of now, some parts of the code seem to be unnecessarily long.

wlren committed:
GUI first iteration set up

wlren committed:
Merge branch 'Level-10'

wlren committed:
Reset progress to redo GUI

wlren committed:
Merge branch 'add-gradle-support' of github.com:wlren/ip into add-gradle-support

wlren committed:
Merge branch 'add-gradle-support'

wlren committed:
Update for Gradle support

wlren committed:
Minor syntax touch up

wlren committed:
Hello World GUI

wlren committed:
GUI Echo iteration

wlren committed:
FXML integration

wlren committed:
Refactor Duke logic for GUI

wlren committed:
Merge branch 'branch-Level-10'

wlren committed:
Code quality - JavaDocs

Large amounts of JavaDocs missing or not according to standard

Global fix of most JavaDocs on new methods from refactoring for JavaFX
and update previously low-quality JavaDocs.

Cleaning up JavaDocs ensures future development to be smoother and
faster.

wlren committed:
Code quality - Refactoring

wlren committed:
Assertion tests on command

wlren committed:
Merge pull request #2 from wlren/branch-A-Assertions

Assertion tests on command

wlren committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Parser.java

wlren committed:
Merge pull request #3 from wlren/branch-A-CodeQuality

Merge code quality change

wlren committed:
Extension for C-Archive

wlren committed:
Merge branch 'branch-BCD-Extension'

Extension for category C-Archive

Provide a way to archive items to that the user can remove items from
the app but still keep a record of them somewhere.

Archived using .txt with timestamp to ensure unique archives and follow
same file standards as normal data store.

wlren committed:
Set theme jekyll-theme-midnight

wlren committed:
UI Screenshot

wlren committed:
Merge branch 'master' of github.com:wlren/ip

wlren committed:
Fix bye and archive

wlren committed:
Main README.md update

wlren committed:
User guide for release
